pandas/types/concat.py

```diff
@@ -201,6 +201,46 @@ def convert_categorical(x):
         return Categorical(concatted, rawcats)
 
 
+def union_categoricals(to_union):
+    """
+    Combine list-like of Categoricals, unioning categories. All
+    must have the same dtype, and none can be ordered.
+
+    Parameters
+    ----------
+    to_union : list like of Categorical
+
+    Returns
+    -------
+    Categorical
+       A single array, categories will be ordered as they
+       appear in the list
+    """
+    from pandas import Index, Categorical
+
+    if any(c.ordered for c in to_union):
+        raise TypeError("Can only combine unordered Categoricals")
+
+    first = to_union[0]
+    if not all(com.is_dtype_equal(c.categories, first.categories)
+               for c in to_union):
+        raise TypeError("dtype of categories must be the same")
+
+    for i, c in enumerate(to_union):
+        if i == 0:
+            cats = c.categories.tolist()
```

Comment:
you soln won't preseve order, though with a small mod I think it could, need to ``np.concatenate`` the categories directly (that's why they were listifed in the first place). Index union doesn't preserve order.